html/videoroomtest.js

```diff
@@ -265,6 +268,45 @@ $(document).ready(function() {
 	}});
 });
 
+function initDevices(devices) {
+	$('#devices').removeClass('hide');
+	var audio = $('#audio-device').val();
+	var video = $('#video-device').val();
+	$('#audio-device, #video-device').find('option').remove();
+
+	devices.forEach(function(device) {
+		var option = $('<option value="' + device.deviceId + '">' + device.label + '</option>');
+		if(device.kind === 'audioinput') {
+			$('#audio-device').append(option);
+		} else if(device.kind === 'videoinput') {
+			$('#video-device').append(option);
+		}
+	});
+
+	$('#audio-device').val(audio);
+	$('#video-device').val(video);
+
+	$('#audio-device, #video-device').on('change', function() {
+		sfutest.send({
+			message: { request: "unpublish" },
+			success: function() {
+				// https://github.com/meetecho/janus-gateway/issues/425
+				setTimeout(function() {
+					publishOwnFeed({
+						deviceId: {
+							exact: $('#audio-device').val()
+						}
+					}, {
+						deviceId: {
+							exact: $('#video-device').val()
+						}
+					});
+				}, 1000);
+			}
+		});	
+	});
+}
+
```

Comment:
Again, if the main purpose of this is to showcase how you can enumerate devices and choose one over another, I wouldn't do this in one of the existing demos. I'd make a new demo page that uses the EchoTest instead (or maybe extend the EchoTest itself, as it's much more evident there as you can see the changes). The VideoRoom demo purpose is to show how you can implement media routing simply, and these additions are distracting IMHO.